include/mbedtls/pkcs7_write.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * \file pkcs7_write.h
+ *
+ * \brief PKCS7 generic defines and structures
+ *  https://tools.ietf.org/html/rfc2315
+ */
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+#ifndef MBEDTLS_PKCS7_WRITE_H
+#define MBEDTLS_PKCS7_WRITE_H
+
+#include "mbedtls/private_access.h"
+
+#include "mbedtls/build_info.h"
+
+#include "md.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \brief                   This function generates a PKCS7 containing SignedData (sect 9.1)
+ *                          as described by RFC 2315 (https://tools.ietf.org/html/rfc2315).
+ *                          It uses the given \p crts and \p keys to sign the \p data and
+ *                          return a valid PKCS7 buffer, \p pkcs7.
+ *
+ * \param pkcs7             The resulting PKCS7, this is the output data.
+ * \param pkcs7_size        The length of \p pkcs7.
+ * \param data              The data that is to be the contents of the PKCS7.
+ *                          This is what gets signed.
+ *                          Pass \c NULL if data length is 0.
+ * \param data_size         Length of \p data.
+ *                          Pass 0 if \p data is \c NULL.
+ * \param crts              Array of x509 certificates (DER).
+ *                          These should have a corresponding private key
+ *                          with same index in \p keys.
+ * \param keys              Array of private keys (DER).
+ * \param crt_sizes         Array of sizes for \p crts . Indexs should match
+ * \param key_sizes         Array of sizes for \p keys.
+ * \param key_pairs         Number or key pairs. Array length of key/crtFiles.
+ * \param hash_funct        Hash function to use in digest, see mbedtls_md_type_t for
+ *                          values in mbedtls/md.h
+ * \param f_rng             RNG function. This must not be \c NULL.
+ * \param p_rng             RNG parameter, can be \c NULL .
+ * \param keys_are_sigs     Flag that is set to 0 if \p keys contains private keys
```

Comment:
Agree that it looks weird. but not sure if separating into two different arguments would anything more clear. It would also require 2 additional arguments: one for optional signatures and one for optional signature sizes. This introduces a couple places where the user could make a mistake when passing arguments. The way it is setup now, the user cannot make an invalid call to the function with arguments alone (not sure if that makes sense).

What are your thoughts on making two new wrapper functions around a `static/private` version of `mbedtls_pkcs7_create`? Something like `mbedtls_pkcs7_create_w_keys` and `mbedtls_pkcs7_create_w_sig` which make calls to `mbedtls_pkcs7_create` under the hood.